src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = Object.assign({}, state);
+  const arrayResult = [];
+
+  for (const item of actions) {
+    switch (item.type) {
+      case 'addProperties': {
+        for (const key in item.extraData) {
+          stateCopy[key] = item.extraData[key];
+        }
+        arrayResult.push({ ...stateCopy });
```

Comment:
move this push after switch case actions to not repeat. You're doing same action for all cases, so it could be moved outside